lib/uhash.ts

```diff
@@ -0,0 +1,54 @@
+// This module implements the "uhash" signature specification described in docs/signature_specification.md
+import * as crypto from 'crypto';
+import { SignatureResult } from './types';
+import { isBinary } from './utils/binary';
+
+const hashAlgorithm = 'md5';
+
+export const getUhashSignature = (
+  filePath: string,
+  fileContents: Buffer,
+): SignatureResult => {
+  const digest = getUhashDigest(fileContents, isBinary(fileContents));
+  const data = digest.slice(0, 24);
+  return {
+    path: filePath,
+    hashes_ffm: [
+      {
+        data,
+        format: 3,
+      },
+    ],
+  };
+};
+
+const getUhashDigest = (fileContents: Buffer, isBinary: boolean): string => {
+  const file = isBinary ? fileContents : removeWhitespaceBytewise(fileContents);
+  if (hashAlgorithm == 'md5') {
+    const hash = crypto.createHash(hashAlgorithm).update(file);
+    return hash.digest('hex');
+  } else {
+    // placeholder for non-md5 hashing algorithm, e.g. xxhash
```

Comment:
maybe throw here or something instead, to make it more visible what the problem is

Reply:
good idea.